sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -52,7 +52,14 @@ class JDBCOptions(
    */
   val asProperties: Properties = {
     val properties = new Properties()
-    parameters.originalMap.foreach { case (k, v) => properties.setProperty(k, v) }
+    parameters.originalMap.foreach { case (k, v) =>
+      // If an option value is `null`, throw a user-friendly error. Keys here cannot be null, as
+      // scala's implementation of Maps prohibits null keys.
+      if (v == null) {
+        throw QueryCompilationErrors.nullArgumentError("JDBC Options", k)
```

Comment:
While I am fine with this, we should really fix Python side together. Python just ignores `None` IIRC. This PR doesn't have to address all the Python cases but it would be great if we can make Python consistent with this.

Reply:
I can followup with python fix for this in separate PR?

Reply:
The problem is that we throw an exception with `None`, then now it's a breaking change. That's why I have been hesitant about doing this incrementally, e.g., https://github.com/apache/spark/pull/46626